source/1_engine/rex_engine/include/rex_engine/string/stringid.h

```diff
@@ -0,0 +1,60 @@
+#pragma once
+
+#include "rex_engine/string/stringentry.h"
+#include "rex_engine/types.h"
+#include "rex_std/ostream.h"
+#include "rex_std/string.h"
+
+namespace rex
+{
+  class StringID
+  {
+  public:
+    StringID();
+    explicit StringID(rsl::string_view characters);
+
+    explicit operator u32() const;
+
+    bool operator==(const StringID& other) const;
+    bool operator!=(const StringID& other) const;
+
+    bool operator==(const StringEntryID& entryID) const;
+    bool operator!=(const StringEntryID& entryID) const;
+
+    rsl::string_view to_string_view() const;
+    bool is_none() const;
+    u32 value() const;
+
+  private:
+    /** Index into the StringID array (used to find String portion of the string/number pair used for comparison) */
+    StringEntryID m_comparison_index;
+  };
+
+  StringID create_sid(rsl::string_view stringView);
+
+  bool operator==(rsl::string_view s, const StringID& sid);
+  bool operator!=(rsl::string_view s, const StringID& sid);
+  bool operator==(const StringID& sid, rsl::string_view s);
+  bool operator!=(const StringID& sid, rsl::string_view s);
+} // namespace rex
+
+rex::StringID operator""_sid(const char* string, size_t size);
+
+rsl::ostream& operator<<(rsl::ostream& os, const rex::StringID& stringID);
```

Comment:
probably best if you create a formatter for this as well.
I'll see if I can create a default formatter to work with rsl::format that uses the operator<< overload for ostream

Reply:
Made an issue for it, so I can figure out how to do this but the branch can already be merged. It's an addition to the API not a change so I think this it's fine to do it after.